### PR TITLE
Gh 130

### PR DESCRIPTION
**Pull request by shadracknx**
### What do these changes do?

1. Updates Android SDK versions to 16.5.1. 
2. Updates  iOS SDK version to 16.7.0.
3. Update to Kotlin 1.7.0
4. Updates flutter plugin version to 6.0.0
5. Fix Flutter Plugin `pubspec.yaml` to follow file conventions 
       * reduced description characters to under 180chars recommended
6. Update version update script to update readme and yaml as well.
7.  Migrate example to flutter 3
      * removed unnecessary swipe to refresh package
      * update android gradle to 7.1.1
9. Updates CHANGELOG.md

### Why are these changes necessary?

1. Takes advantages of Airship latest SDK
2. Fix pub points on pub.dev
3. Improve build tools.

### How did you verify these changes?

I run the example app, and nothing broke; no warning and everything works as expected on IOS and android Emulators.

### Anything else a reviewer should know?

1. My testing for what changed in Airship native  was non-existent, my focus was on flutter.
2. Am not sure what this version breaks for existing code based wanting to upgrade except for a requirement of full upgrade to flutter 3; there is no migration doc provided.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Flutter Plugin Changelog
 
+## Version 5.6.0 - June 13, 2022
+Minor release that updates Airship Android SDK to 16.5.1, iOS SDK to 16.7.0,
```
Comment (rlepinski) [inline CHANGELOG.md:4]: Should be major. Major release that support Flutter 3.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,14 @@
 # Flutter Plugin Changelog
 
+## Version 5.6.0 - June 13, 2022
```
Comment (rlepinski) [inline CHANGELOG.md:3]: 6.0?

Comment (shadracknx) [reply CHANGELOG.md]: updated to 6.0.0

File: android/build.gradle [hunk 3]
```diff
@@ -2,9 +2,9 @@ group 'com.airship.airship'
 version '1.0-SNAPSHOT'
 
 buildscript {  
-    ext.kotlin_version = '1.5.31'
-    ext.coroutine_version = '1.5.0'
-    ext.airship_version = '16.4.0'
+    ext.kotlin_version = '1.7.0'
```
Comment (jyaganeh) [inline android/build.gradle:5]: we haven't bumped Kotlin in the Android SDK yet, so it'd probably be best to keep kotlin on `1.5.31` and bump coroutines to `1.5.2` to match for now

File: example/android/build.gradle [hunk 4]
```diff
@@ -1,5 +1,5 @@
 buildscript {
-    ext.kotlin_version = '1.5.31'
+    ext.kotlin_version = '1.7.0'
```
Comment (jyaganeh) [inline example/android/build.gradle:2]: this is probably fine to update, but look out for warnings in the gradle logs that mention multiple kotlin versions being on the classpath, which may happen if this version doesn't match what we've declared in the other build.gradle for the library

**Review by jyaganeh — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,16 @@
 # Flutter Plugin Changelog
 
+## Version 6.0.0 - June 13, 2022
+Minor release that supports flutter 3.
```
Comment (rlepinski) [inline CHANGELOG.md:4]: Major*

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,16 @@
...
+
+- updates Airship Android SDK to 16.5.1, 
+- Updates iOS SDK to 16.7.0,
```
Comment (rlepinski) [inline CHANGELOG.md:7]: `- Updated Airship iOS SDK to 16.7.0.`

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,16 @@
 # Flutter Plugin Changelog
 
+## Version 6.0.0 - June 13, 2022
+Minor release that supports flutter 3.
+
+- updates Airship Android SDK to 16.5.1, 
+- Updates iOS SDK to 16.7.0, 
+- Flutter SDK to 3.0.2 and dart SDK 2.17.3
+- Fixed code flutter formatting
```
Comment (rlepinski) [inline CHANGELOG.md:9]: Add periods to the end

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,16 @@
 # Flutter Plugin Changelog
 
+## Version 6.0.0 - June 13, 2022
+Major release that supports flutter 3.
+
+- updated Airship Android SDK to 16.5.1.
```
Comment (jyaganeh) [inline CHANGELOG.md:6]: the first `u` could be capitalized here

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,16 @@
...
+- Updated Airship iOS SDK to 16.7.0.
+- Updated Flutter SDK to 3.0.2 and dart SDK 2.17.3.
+- Fixed code flutter formatting.
+- Fixed pubspec to follow Dart file conventions.
+- updated Flutter example to flutter 3.
```
Comment (jyaganeh) [inline CHANGELOG.md:11]: capitalize `u` here, too

File: pubspec.yaml [hunk 10]
```diff
@@ -1,13 +1,14 @@
 name: airship_flutter
-description: "The Airship flutter plugin. This package implements a cross-platform plugin interface to Airship's iOS and Android native SDKs. This allows core Airship functionality to be implemented by Flutter apps written in dart. The available core Airship functionality currently includes: push, in-app message, message center, actions, custom events and more."
-version: 5.5.0
+description: "The Airship flutter cross-platform plugin interface to Airship's iOS and Android native SDKs.
+Simplifies adding core Airship support to be Flutter dart apps."
```
Comment (jyaganeh) [inline pubspec.yaml:3]: we could maybe shorten this to something like the following, assuming it meets the min/max requirements on pub.dev

> Cross-platform plugin interface for the native Airship iOS and Android SDKs. Simplifies adding Airship to Flutter apps.

File: .github/workflows/release.yml [hunk 11]
```diff
@@ -41,14 +47,15 @@ jobs:
...
-      - uses: actions/setup-java@v1
+      - uses: actions/setup-java@v2
         with:
-          java-version: "12.x"
+          distribution: ${{ env.JAVA_DISTRIBUTION }}
+          java-version: ${{ env.JAVA_VERSION }}
 
       - name: Flutter action
-        uses: subosito/flutter-action@4389e6cbc6cb8a4b18c628ff96ff90be0e926aa8
+        uses: subosito/flutter-action@v2
```
Comment (jyaganeh) [inline .github/workflows/release.yml:56]: hash instead of `v2` tag

File: .github/workflows/release.yml [hunk 12]
```diff
@@ -89,22 +96,23 @@ jobs:
...
```
Comment (jyaganeh) [inline .github/workflows/release.yml:113]: hash instead of `v2` here, also

File: .github/workflows/release.yml [hunk 13]
```diff
@@ -5,27 +5,33 @@ on:
...
-      - uses: actions/checkout@v2
-      - uses: actions/setup-java@v1
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v2
         with:
-          java-version: "12.x"
-      - uses: subosito/flutter-action@4389e6cbc6cb8a4b18c628ff96ff90be0e926aa8
+          distribution: ${{ env.JAVA_DISTRIBUTION }}
+          java-version: ${{ env.JAVA_VERSION }}
+      - uses: subosito/flutter-action@v2
```
Comment (jyaganeh) [inline .github/workflows/release.yml:24]: use the hash instead of `v2` tag here

File: .github/workflows/ci.yml [hunk 14]
```diff
@@ -1,48 +1,52 @@
...
-      - uses: actions/checkout@v1
-      - uses: actions/setup-java@v1
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v2
         with:
-          java-version: "12.x"
-      - uses: subosito/flutter-action@4389e6cbc6cb8a4b18c628ff96ff90be0e926aa8
+          distribution: ${{ env.JAVA_DISTRIBUTION }}
+          java-version: ${{ env.JAVA_VERSION }}
+      - uses: subosito/flutter-action@v2
```
Comment (jyaganeh) [inline .github/workflows/ci.yml:19]: For 3rd-party actions, we prefer commit hashes instead of version tags, since that makes it less likely for workflow logic to be changed out from under us. It's probably not all that likely that a maintainer or bad actor would do that and actually impact our jobs, but better to be safe than sorry. 🙏

File: .github/workflows/ci.yml [hunk 15]
```diff
@@ -1,48 +1,52 @@
...
```
Comment (jyaganeh) [inline .github/workflows/ci.yml:48]: this one should also be a hash instead of the `v2` tag

File: scripts/update_version.sh [hunk 16]
```diff
@@ -11,9 +12,14 @@ if [ -z "$1" ]
...
-sed -i '' "s/\(AIRSHIP_PLUGIN_VERSION *= *\)\".*\"/\1\"$VERSION\"/g" $ROOT_PATH/android/src/main/kotlin/com/airship/flutter/AirshipPluginVersion.kt
+sed -i '' "s/\(pluginVersion *= *\)\".*\"/\1\"$VERSION\"/g" "$ROOT_PATH/ios/Classes/AirshipPluginVersion.swift"
+sed -i '' "s/\(AIRSHIP_PLUGIN_VERSION *= *\)\".*\"/\1\"$VERSION\"/g" "$ROOT_PATH/android/src/main/kotlin/com/airship/flutter/AirshipPluginVersion.kt"
 
 # Update podspec
-sed -i '' "s/\(^AIRSHIP_FLUTTER_VERSION *= *\)\".*\"/\1\"$VERSION\"/g" $ROOT_PATH/ios/airship_flutter.podspec
+sed -i '' "s/\(^AIRSHIP_FLUTTER_VERSION *= *\)\".*\"/\1\"$VERSION\"/g" "$ROOT_PATH/ios/airship_flutter.podspec"
 
+# Update version in README.md with the current
+sed -i '' "s/\(^  airship_flutter *: *\).*/\1\^$VERSION/g" "$ROOT_PATH/README.md"
```
Comment (jyaganeh) [inline scripts/update_version.sh:22]: Thanks for updating this! 🙌

**Review by jyaganeh — APPROVED** [review]
left a few relatively minor non-code comments, but this lgtm overall!

File: pubspec.yaml (hunk 10 above)
Comment (shadracknx) [reply pubspec.yaml:3]: I like that

File: scripts/update_version.sh (hunk 16 above)
Comment (shadracknx) [reply scripts/update_version.sh:22]: Glad you like.

File: CHANGELOG.md (hunk 9 above)
Comment (shadracknx) [reply CHANGELOG.md:11]: fixed
